service/xmpp/XMPPEvents.ts

```diff
@@ -0,0 +1,484 @@
+export const READY_TO_JOIN = XMPPEvents.READY_TO_JOIN;
+export const RECORDER_STATE_CHANGED = XMPPEvents.RECORDER_STATE_CHANGED;
+export const REMOTE_STATS = XMPPEvents.REMOTE_STATS;
+export const RENEGOTIATION_FAILED = XMPPEvents.RENEGOTIATION_FAILED;
+export const RESERVATION_ERROR = XMPPEvents.RESERVATION_ERROR;
+export const ROOM_CONNECT_ERROR = XMPPEvents.ROOM_CONNECT_ERROR;
+export const ROOM_CONNECT_NOT_ALLOWED_ERROR = XMPPEvents.ROOM_CONNECT_NOT_ALLOWED_ERROR;
+export const ROOM_JOIN_ERROR = XMPPEvents.ROOM_JOIN_ERROR;
+export const ROOM_CONNECT_MEMBERS_ONLY_ERROR = XMPPEvents.ROOM_CONNECT_MEMBERS_ONLY_ERROR;
+export const ROOM_MAX_USERS_ERROR = XMPPEvents.ROOM_MAX_USERS_ERROR;
+export const SENDING_CHAT_MESSAGE = XMPPEvents.SENDING_CHAT_MESSAGE;
+export const SENDING_PRIVATE_CHAT_MESSAGE = XMPPEvents.SENDING_PRIVATE_CHAT_MESSAGE;
+export const SESSION_ACCEPT = XMPPEvents.SESSION_ACCEPT;
+export const SESSION_ACCEPT_ERROR = XMPPEvents.SESSION_ACCEPT_ERROR;
+export const SESSION_ACCEPT_TIMEOUT = XMPPEvents.SESSION_ACCEPT_TIMEOUT;
+export const SOURCE_ADD = XMPPEvents.SOURCE_ADD;
+export const SOURCE_ADD_ERROR = XMPPEvents.SOURCE_ADD_ERROR;
+export const SOURCE_REMOVE = XMPPEvents.SOURCE_REMOVE;
+export const SOURCE_REMOVE_ERROR = XMPPEvents.SOURCE_REMOVE_ERROR;
+export const SPEAKER_STATS_RECEIVED = XMPPEvents.SPEAKER_STATS_RECEIVED;
+export const CONFERENCE_TIMESTAMP_RECEIVED = XMPPEvents.CONFERENCE_TIMESTAMP_RECEIVED;
+export const AV_MODERATION_APPROVED = XMPPEvents.AV_MODERATION_APPROVED;
+export const AV_MODERATION_REJECTED = XMPPEvents.AV_MODERATION_REJECTED;
+export const AV_MODERATION_RECEIVED = XMPPEvents.AV_MODERATION_RECEIVED;
+export const AV_MODERATION_CHANGED = XMPPEvents.AV_MODERATION_CHANGED;
+export const AV_MODERATION_PARTICIPANT_APPROVED = XMPPEvents.AV_MODERATION_PARTICIPANT_APPROVED;
+export const AV_MODERATION_PARTICIPANT_REJECTED = XMPPEvents.AV_MODERATION_PARTICIPANT_REJECTED;
+export const BREAKOUT_ROOMS_MOVE_TO_ROOM = XMPPEvents.BREAKOUT_ROOMS_MOVE_TO_ROOM;
+export const BREAKOUT_ROOMS_EVENT = XMPPEvents.BREAKOUT_ROOMS_EVENT;
+export const BREAKOUT_ROOMS_UPDATED = XMPPEvents.BREAKOUT_ROOMS_UPDATED;
+export const START_MUTED_FROM_FOCUS = XMPPEvents.START_MUTED_FROM_FOCUS;
+export const SUBJECT_CHANGED = XMPPEvents.SUBJECT_CHANGED;
+export const SUSPEND_DETECTED = XMPPEvents.SUSPEND_DETECTED;
+export const TRANSCRIPTION_STATUS_CHANGED = XMPPEvents.TRANSCRIPTION_STATUS_CHANGED;
+export const TRANSPORT_INFO = XMPPEvents.TRANSPORT_INFO;
+export const VIDEO_SIP_GW_AVAILABILITY_CHANGED = XMPPEvents.VIDEO_SIP_GW_AVAILABILITY_CHANGED;
+export const VIDEO_SIP_GW_SESSION_STATE_CHANGED = XMPPEvents.VIDEO_SIP_GW_SESSION_STATE_CHANGED;
+export const ICE_CONNECTION_STATE_CHANGED = XMPPEvents.ICE_CONNECTION_STATE_CHANGED;
+export const JSON_MESSAGE_RECEIVED = XMPPEvents.JSON_MESSAGE_RECEIVED;
+
+// TODO: this was a pre-ES6 module using module.exports = XMPPEvents which doesn't translate well
```

Comment:
How about we do this in this very PR? I can trigger a full e2e test to make sure we haven't broken anything.

Reply:
We could do.

My aim in this round was to ensure 100% backward compatibility without breaking any consumers of these files - you know a great deal more than I do about whether these are used by others or whether they are effectively "internal" code.

My personal preference would be to do these in two stages - get the TS merged in - and then for me to work through the TODO's I've added with PRs that cause breaking changes. These then get evaluated in their own right.

Flip of a coin stuff isn't it? There are advantages to both approaches.

Reply:
Up until very recently the only way this library was consumed was through the webpack bundle. Now there is a tsc build, which mobile uses, but we export a single object with everything, and AFAICT we can keep all of that working just fine and change the internals. That's fair game.

Reply:
Yeah - that's the sort of knowledge that you have that I haven't the slightest about :-)

So is the decision to make the changes in this PR?

Reply:
Yeah let's do it!

PS: Very happy with the work you're doing, thank you! ❤️

Reply:
OK - that's done - but boy it now affects a lot of files - my concern with a branch like this is it becomes much less transparent what the change is due to it affecting so many places

Reply:
Let's see what the tests have to say :-)